book/src/03_3_2_http_client.md

```diff
@@ -12,78 +10,63 @@ In this exercise, we'll write a small client that retrieves data over a HTTP con
 $ cargo doc --open
 ```
 
-✅ Add your network credentials to the `cfg.toml`.
+✅ Add your [network credentials](02_4_hello_board.md) to the `cfg.toml` as previously.
 
 ✅ Open the prepared project skeleton in `intro/http-client/exercise`. We'll read through the code together.
 
 ## Making a connection
 
 By default only unencrypted HTTP is available, which rather limits our options of hosts to connect to. We're going to use `http://neverssl.com/`.
 
-In `esp-idf`, HTTP client connections are managed by `http::client::EspHttpClient` in the `esp-idf-svc` crate. It implements the `http::client::Client` trait from `embedded-svc`, which defines functions for [HTTP request methods](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods) like `GET` or `POST`.
+In `esp-idf`, HTTP client connections are managed by `http::client::EspHttpClient` in the `esp-idf-svc` crate. It implements the `http::client::Client` trait from `embedded-svc`, which defines functions for [HTTP request methods](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods) like `GET` or `POST`. This is a good time to have a look at the documentation you opened with `cargo doc --open`.
+
+✅ Add the url `http://neverssl.com/` is in the main function. This is the address we will query.
 
-✅ Create a new `EspHttpClient` with default values.
+✅ Create a new `EspHttpClient` with default values (Try to find the method `new_default()` in the documentation).
 
-TODO currently no `docs.rs` documentation for the svc crates (ping Espressif about this, it's a known issue though)
 
-Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. This makes more sense with `POST` and `PUT` but must still be performed with `GET`. 
+Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. 
 
 After this optional send step the `Writer` can be turned into a `Response` from which the received server output can be read:
 
-TODO explain as_ref(). Or find a better way to point people to it. 
```

Comment:
Can you revert the changes around the as_ref() topic?
You can keep the your solution as an additional alternative way.

Reply:
I did not change the `solution/main.rs`, the solution without `as_ref()` is in `solution1.rs` for the moment.